cmd/swctl/cmd_status.go

```diff
@@ -34,23 +64,240 @@ func NewStatusCmd(cli Cli) *cobra.Command {
 			UnknownFlags: true,
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			opts.Args = args
 			return runStatusCmd(cli, opts)
 		},
 	}
+	opts.InstallFlags(cmd.PersistentFlags())
 	return cmd
 }
 
-func runStatusCmd(cli Cli, opts StatusCmdOptions) error {
-	cmd := fmt.Sprintf("vpp-probe --env=%s discover", defaultVppProbeEnv)
-	out, err := cli.Exec(cmd, opts.Args)
-	if err != nil {
-		if ee, ok := err.(*exec.ExitError); ok {
-			return fmt.Errorf("%v: %s", ee.String(), ee.Stderr)
+type statusInfo struct {
+	client.Component
+	ConfigCounts configCounts
+}
+
+func runStatusCmd(cli Cli, opts StatusOptions) error {
+	if opts.ShowInterfaces {
+		cmd := fmt.Sprintf("vpp-probe --env=%s --query label=%s=stonework discover", defaultVppProbeEnv, client.DockerComposeServiceLabel)
+		formatArg := fmt.Sprintf("--format=%s", opts.Format)
+		out, err := cli.Exec(cmd, []string{formatArg})
+		if err != nil {
+			if ee, ok := err.(*exec.ExitError); ok {
+				return fmt.Errorf("%v: %s", ee.String(), ee.Stderr)
+			}
 		}
+		fmt.Fprintln(cli.Out(), out)
+		return nil
+	}
+
+	resp, err := cli.Client().GetComponents()
+	if err != nil {
 		return err
 	}
 
-	fmt.Fprintln(cli.Out(), out)
+	type infoWithErr struct {
+		statusInfo
+		error
+	}
+	var infos []statusInfo
+	var wg sync.WaitGroup
+	infoCh := make(chan infoWithErr)
+
+	for _, compo := range resp {
+		wg.Add(1)
+		go func(compo client.Component) {
+			defer wg.Done()
+			var counts configCounts
+			if compo.GetMode() != client.ComponentForeign {
+				vals, err := compo.SchedulerValues()
+				if err != nil {
+					infoCh <- infoWithErr{error: err}
+				}
+				counts = countConfig(vals)
+			}
+			infoCh <- infoWithErr{
+				statusInfo: statusInfo{
+					Component:    compo,
+					ConfigCounts: counts,
+				},
+			}
+		}(compo)
+	}
+
+	go func() {
+		wg.Wait()
+		close(infoCh)
+	}()
+
+	for i := range infoCh {
+		if i.error != nil {
+			return i.error
+		}
+		infos = append(infos, i.statusInfo)
+	}
+	slices.SortFunc(infos, cmpStatus)
+	if opts.Format == "" {
+		printStatusTable(cli.Out(), infos)
+	} else {
+		if err := formatAsTemplate(cli.Out(), opts.Format, infos); err != nil {
+			return err
+		}
+	}
 	return nil
 }
+
+func countConfig(baseVals []*kvscheduler.BaseValueStatus) configCounts {
```

Comment:
This low-level counting should happen in client package. Try to imagine users using the client package directly in their apps. They would have to copy-paste this code to get the same information about component.

Reply:
Put it instead of the `SchedulerValues` function into the component API.